File: app_manager/appmanager/src/main/java/app/editors/manager/mvp/views/main/DocsOnDeviceView.java
package app.editors.manager.mvp.views.main;

import app.editors.manager.mvp.presenters.main.OpenState;
import moxy.viewstate.strategy.OneExecutionStateStrategy;
import moxy.viewstate.strategy.StateStrategyType;
import android.net.Uri;


import java.util.List;

import app.editors.manager.mvp.models.explorer.Explorer;
import app.editors.manager.mvp.models.explorer.Item;

@StateStrategyType(OneExecutionStateStrategy.class)
public interface DocsOnDeviceView extends DocsBaseView {
    void onActionDialog();
    void onRemoveItem(Item item);
    void onRemoveItems(List<Item> items);
    void onShowCamera(Uri photoUri);
    void onShowFolderChooser();
    void onOpenMedia(OpenState.Media state);

    //Open file
    void onShowDocs(Uri uri);
    void onShowCells(Uri uri);
    void onShowSlides(Uri uri);
    void onShowPdf(Uri uri);
}

File: app_manager/appmanager/src/main/java/app/editors/manager/mvp/views/main/DocsOnDeviceView.kt
package app.editors.manager.mvp.views.main

import android.net.Uri
import app.documents.core.network.manager.models.explorer.Item
import app.editors.manager.mvp.presenters.main.OpenState
import lib.toolkit.base.managers.utils.EditType
import moxy.viewstate.strategy.OneExecutionStateStrategy
import moxy.viewstate.strategy.StateStrategyType

@StateStrategyType(OneExecutionStateStrategy::class)
interface DocsOnDeviceView : DocsBaseView {
    fun onActionDialog()
    fun onRemoveItems(vararg items: Item)
    fun onShowPortals()
    fun onOpenMedia(state: OpenState.Media)

    //Open file
    fun onShowPdf(uri: Uri, editType: EditType)
}